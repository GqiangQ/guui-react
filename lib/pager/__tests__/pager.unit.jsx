import {mount} from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'
import Pager from '../pager'


describe(Pager.displayName, () => {
    it('DisplayName', () => {
        expect(Pager.displayName).toEqual('Pager')
    })
    it('prop current = 1', () => {
        const tree = renderer
            .create(<Pager current={1} total={3}></Pager>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('prop total = 5', () => {
        const tree = renderer
            .create(<Pager current={2} total={5}></Pager>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('应该要有两个省略号', () => {
        const tree = renderer
            .create(<Pager current={5} total={10}></Pager>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('应该要有一个省略号', () => {
        const tree = renderer
            .create(<Pager current={8} total={10}></Pager>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('点击对应按钮 current发生对应变化', () => {
        const c = mount(<Pager current={1} total={5} onChange={ (value)=>{c.setProps({current: value})} }/>);
        expect(c.find('.gu-pager-active').text()).toEqual("1")
        c.find('.gu-pager-item').last().simulate('click')
        expect(c.find('.gu-pager-active').text()).toEqual("5")
    });
    it('click prev', () => {
        const c = mount(<Pager current={2} total={5}
                               onChange={ (value)=>{c.setProps({current: value})} }/>);
        c.find('.gu-pager-prev').simulate('click');
        expect(c.find('.gu-pager-active').text()).toEqual("1")
    });
    it('click next', () => {
        const c = mount(<Pager current={2} total={5}
                               onChange={ (value)=>{c.setProps({current: value})} }/>)
        c.find('.gu-pager-next').simulate('click')
        expect(c.find('.gu-pager-active').text()).toEqual("3")
    });
    it('click prev ...', () => {
        const c = mount(<Pager current={5} total={10}
                               onChange={ (value)=>{c.setProps({current: value})} }/>)
        c.find('.gu-pager-separator').first().simulate('click')
        expect(c.find('.gu-pager-active').text()).toEqual("1")
    });
    it('click next ...', () => {
        const c = mount(<Pager current={6} total={100}
                               onChange={ (value)=>{c.setProps({current: value})} }/>)
        c.find('.gu-pager-separator').last().simulate('click')
        expect(c.find('.gu-pager-active').text()).toEqual("11")
    });
})
