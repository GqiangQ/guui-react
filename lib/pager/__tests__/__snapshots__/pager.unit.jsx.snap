// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pager prop current = 1 1`] = `
<div
  className="gu-pager"
>
  <button
    className="gu-pager-prev"
    onClick={[Function]}
  >
    <svg
      className="gu-icon gu-icon-left "
      style={
        Object {
          "fill": undefined,
        }
      }
    >
      <use
        xlinkHref="#left"
      />
    </svg>
  </button>
  <button
    className="gu-pager-item gu-pager-active"
    onClick={[Function]}
  >
    1
  </button>
  <button
    className="gu-pager-item"
    onClick={[Function]}
  >
    2
  </button>
  <button
    className="gu-pager-item"
    onClick={[Function]}
  >
    3
  </button>
  <button
    className="gu-pager-next"
    onClick={[Function]}
  >
    <svg
      className="gu-icon gu-icon-right "
      style={
        Object {
          "fill": undefined,
        }
      }
    >
      <use
        xlinkHref="#right"
      />
    </svg>
  </button>
</div>
`;

exports[`Pager prop total = 5 1`] = `
<div
  className="gu-pager"
>
  <button
    className="gu-pager-prev"
    onClick={[Function]}
  >
    <svg
      className="gu-icon gu-icon-left "
      style={
        Object {
          "fill": undefined,
        }
      }
    >
      <use
        xlinkHref="#left"
      />
    </svg>
  </button>
  <button
    className="gu-pager-item"
    onClick={[Function]}
  >
    1
  </button>
  <button
    className="gu-pager-item gu-pager-active"
    onClick={[Function]}
  >
    2
  </button>
  <button
    className="gu-pager-item"
    onClick={[Function]}
  >
    3
  </button>
  <button
    className="gu-pager-item"
    onClick={[Function]}
  >
    4
  </button>
  <button
    className="gu-pager-item"
    onClick={[Function]}
  >
    5
  </button>
  <button
    className="gu-pager-next"
    onClick={[Function]}
  >
    <svg
      className="gu-icon gu-icon-right "
      style={
        Object {
          "fill": undefined,
        }
      }
    >
      <use
        xlinkHref="#right"
      />
    </svg>
  </button>
</div>
`;

exports[`Pager 应该要有一个省略号 1`] = `
<div
  className="gu-pager"
>
  <button
    className="gu-pager-prev"
    onClick={[Function]}
  >
    <svg
      className="gu-icon gu-icon-left "
      style={
        Object {
          "fill": undefined,
        }
      }
    >
      <use
        xlinkHref="#left"
      />
    </svg>
  </button>
  <button
    className="gu-pager-item"
    onClick={[Function]}
  >
    1
  </button>
  <span
    className="gu-pager-separator"
    onClick={[Function]}
  >
    ...
  </span>
  <button
    className="gu-pager-item"
    onClick={[Function]}
  >
    6
  </button>
  <button
    className="gu-pager-item"
    onClick={[Function]}
  >
    7
  </button>
  <button
    className="gu-pager-item gu-pager-active"
    onClick={[Function]}
  >
    8
  </button>
  <button
    className="gu-pager-item"
    onClick={[Function]}
  >
    9
  </button>
  <button
    className="gu-pager-item"
    onClick={[Function]}
  >
    10
  </button>
  <button
    className="gu-pager-next"
    onClick={[Function]}
  >
    <svg
      className="gu-icon gu-icon-right "
      style={
        Object {
          "fill": undefined,
        }
      }
    >
      <use
        xlinkHref="#right"
      />
    </svg>
  </button>
</div>
`;

exports[`Pager 应该要有两个省略号 1`] = `
<div
  className="gu-pager"
>
  <button
    className="gu-pager-prev"
    onClick={[Function]}
  >
    <svg
      className="gu-icon gu-icon-left "
      style={
        Object {
          "fill": undefined,
        }
      }
    >
      <use
        xlinkHref="#left"
      />
    </svg>
  </button>
  <button
    className="gu-pager-item"
    onClick={[Function]}
  >
    1
  </button>
  <span
    className="gu-pager-separator"
    onClick={[Function]}
  >
    ...
  </span>
  <button
    className="gu-pager-item"
    onClick={[Function]}
  >
    3
  </button>
  <button
    className="gu-pager-item"
    onClick={[Function]}
  >
    4
  </button>
  <button
    className="gu-pager-item gu-pager-active"
    onClick={[Function]}
  >
    5
  </button>
  <button
    className="gu-pager-item"
    onClick={[Function]}
  >
    6
  </button>
  <button
    className="gu-pager-item"
    onClick={[Function]}
  >
    7
  </button>
  <span
    className="gu-pager-separator"
    onClick={[Function]}
  >
    ...
  </span>
  <button
    className="gu-pager-item"
    onClick={[Function]}
  >
    10
  </button>
  <button
    className="gu-pager-next"
    onClick={[Function]}
  >
    <svg
      className="gu-icon gu-icon-right "
      style={
        Object {
          "fill": undefined,
        }
      }
    >
      <use
        xlinkHref="#right"
      />
    </svg>
  </button>
</div>
`;
